### PR TITLE
To align hive plan with native plan

### PR DESCRIPTION
**Pull request by dirtysalt**
This PR is to align hive plan with native plan. 

There are many unique traits of native plan:
- colocate join
- bucket join
- replicated join
- tablets number in scan node(which will affect cost of broadcast)

To compare hive and native plan in an apple-to-apple way, we have to 
1. disable colocate and bucket and replicated join (`disable_colocate_join=true,cbo_enable_replicated_join=false,disable_bucket_join=true`)
2. tablets number of hive scan operator is large enough, otherwise the cost of broadcast will be underestimated.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ChildPropertyDeriver.java [hunk 1]
```diff
@@ -425,6 +425,10 @@ private void tryReplicatedHashJoin(ExpressionContext context,
      * */
     private void tryBucketShuffle(PhysicalHashJoinOperator node, HashDistributionSpec leftShuffleDistribution,
                                   HashDistributionSpec rightShuffleDistribution) {
+        if (ConnectContext.get().getSessionVariable().isDisableColocateJoin()) {
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ChildPropertyDeriver.java:428]: so we can disable bucket shuffle and disable colocate join at the same time. to align hive plan and native plan, we need to disable both.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java [hunk 2]
```diff
@@ -246,7 +246,7 @@ public CostEstimate visitPhysicalHashAggregate(PhysicalHashAggregateOperator nod
             Statistics inputStatistics = context.getChildStatistics(0);
             CostEstimate otherExtraCost = computeAggFunExtraCost(node, statistics, inputStatistics);
             return CostEstimate.addCost(CostEstimate.of(inputStatistics.getComputeSize(),
-                    CostEstimate.isZero(otherExtraCost) ? statistics.getComputeSize() : 0, 0),
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java:249]: just format problem.

File: fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java [hunk 3]
```diff
@@ -64,4 +64,6 @@
     public static String null_string = "\\N";
 
     public static boolean USE_MOCK_DICT_MANAGER = false;
+
+    public static final int default_tablet_number = 128;
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java:68]: Could you explain why we choose `128` except `256` or `64`?

Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java:68]: Quite hard. It's just a rule-of-thumb. But the reason not to set this value too low is


- The cost of broadcast join, is to `parallelExecInstanceNum* beNum * outputSize`
- `parallelExecInstanceNum` is determined by `getLeftMostScanTabletsNum`(which is tablets number the scan operator scans)
- In most case, the `parallelExecInstanceNum` is actually `ConnectContext.get().getSessionVariable().getDegreeOfParallelism()`, so we want to set tablet number high enough.

```
        @Override
        public CostEstimate visitPhysicalDistribution(PhysicalDistributionOperator node, ExpressionContext context) {
                case BROADCAST:
                    if (statistics.getOutputSize(outputColumns) > sessionVariable.getMaxExecMemByte()) {
                        return CostEstimate.infinite();
                    }
                    int parallelExecInstanceNum = Math.max(1, getParallelExecInstanceNum(context));
                    // beNum is the number of right table should broadcast, now use alive backends
                    int beNum = Math.max(1, Catalog.getCurrentSystemInfo().getBackendIds(true).size());
                    result = CostEstimate
                            .of(statistics.getOutputSize(outputColumns) *
                                            Catalog.getCurrentSystemInfo().getBackendIds(true).size(),
                                    statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum,
                                    statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum);
                    break;

...


        private int getParallelExecInstanceNum(ExpressionContext context) {
            return Math.min(ConnectContext.get().getSessionVariable().getDegreeOfParallelism(),
                    context.getRootProperty().getLeftMostScanTabletsNum());
        }
```

Comment (DorianZheng) [reply fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java:68]: @imay @kangkaisen PTAL

**Review by Youngwb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 4]
```diff
@@ -296,6 +298,9 @@
     @VariableMgr.VarAttr(name = DISABLE_COLOCATE_JOIN)
     private boolean disableColocateJoin = false;
 
+    @VariableMgr.VarAttr(name = DISABLE_BUCKET_JOIN)
+    private boolean disableBucketJoin = false;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:302]: Make this variable to be INVISIBLE, we shouldn't export it to user.

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**